_XOPEN_SOURCE 700 is unknown in 11.3; use _XOPEN_SOURCE 600/XPG6 instead

__EXTENSIONS__ is required for termios IMAXBEL

--- a/meson.build	2023-03-30 18:02:19.434422873 +0000
+++ b/meson.build	2023-03-30 18:02:19.478837442 +0000
@@ -216,8 +216,8 @@
 libc_feature_defines = [
   ['_GNU_SOURCE', '1'],
   ['_POSIX_C_SOURCE', '200809L'],
-  ['_XOPEN_SOURCE', '700'],
-  ['_XOPEN_SOURCE_EXTENDED', '1'],
+  ['_XOPEN_SOURCE', '600'],
+  ['__EXTENSIONS__', '1'],
 ]
 
 system = host_machine.system()
